### ISSUE TITLE
switch_case_on_newline should support `where` in a new line

### ISSUE THREAD

**Issue by nekrich** [realm/SwiftLint#1268]
Example:
```swift
enum SampleError: Error, Equatable {
	
	case boom
	case bang
	case ouch
	case badaboom(code: Int, description: String)
	
	static func == (lhs: SampleError, rhs: SampleError) -> Bool {
		
		switch (lhs, rhs) {
			
		case let (.badaboom(code: lhsErrorCode, description: _), .badaboom(code: rhsErrorCode, description: _))
			where lhsErrorCode == rhsErrorCode:
			
			return true
			
		case (.boom, .boom),
		     (.bang, .bang), (.ouch, .ouch):
			
			return true
			
		default:
			
			return false
			
		}
		
	}
	
}
```

1. false positive in first case - `case let (.badaboom(...`.
2. no warning on the second line of second case: `(.bang, .bang), (.ouch, .ouch):`.

swiftlint version: 0.16.1

**Comment by marcelofabri** [comment]
I could reproduce 1, but I'm not sure what you mean with 2. It seems to be the expected behavior. 🤔

**Comment by marcelofabri** [comment]
BTW, I don't think the issue is about enum with tuples, but breaking the case in 2 lines when using `where`.

**Comment by nekrich** [comment]
I think, that is my fail in understanding that rule.
I just prefer:
a) to group list of values in one `case` statement where it acceptable and in this case to write each pattern from a new line.
b) and write `where` conditions from a new line (not only in `switch` statement).

**Comment by marcelofabri** [comment]
I'll reopen since I think we should support 1 anyway.

Just to be clear: the rule only validates that the body of a case is in a new line.
